tasks/snapshot.yml

```diff
@@ -0,0 +1,35 @@
+version: '3'
+
+tasks:
+  directory:
+    desc: "Creates a snapshot of the current working directory with .git, files in .drainpipeignore, and .drainpipeignore excluded. The target directory should be provided as a cli argument e.g. task snapshot:directory -- /tmp/release"
+    cmds:
+      - if [ -d "{{.CLI_ARGS}}" ] || [ "" == "{{.CLI_ARGS}}" ]; then echo "Please provide a path to a directory that does not yet exist" && exit 1; fi
+      - cp -R . {{.CLI_ARGS}}
+      - if [ ! -f ".drainpipeignore" ]; then echo ".drainpipeignore does not exist" && touch {{.CLI_ARGS}}/.drainpipeignore; fi
+      - |
+        cd {{.CLI_ARGS}}
+        rm -rf .git
+        git init
+        mv .drainpipeignore .gitignore
+        git add .
+        git clean -ffdX
+        rm -rf .git
+  archive:
+    silent: true
+    desc: "Creates a snapshot of the current working directory as an archive. Format options are tar, tar.gz, or zip. Usage: task snapshot:archive format=zip > ~/archive.zip"
+    cmds:
+      - |
+        TMP_DIR=$(mktemp -d)
+        cp -R . $TMP_DIR
+        if [ ! -f ".drainpipeignore" ]; then echo ".drainpipeignore does not exist" && touch $TMP_DIR/.drainpipeignore; fi
+        cd $TMP_DIR
+        rm -rf .git
+        git init
+        mv .drainpipeignore .gitignore
+        git add .
+        git clean -ffdX
+        git config user.email 'no-reply@lullabot.com'
+        git config user.name 'Lullabot Drainpipe'
+        git commit -m "Initial commit" --no-gpg-sign
+        git archive --format={{.format}} HEAD
```

Comment:
From `git archive --help`:

> git archive behaves differently when given a tree ID versus when given
>        a commit ID or tag ID. In the first case the current time is used as
>        the modification time of each file in the archive. In the latter case
>        the commit time as recorded in the referenced commit object is used
>        instead.

Does that mean we're losing modification timestamps since we're creating a whole new repository?

In `ATTRIBUTES export-ignore`: does that mean that `.gitattributes` could also change what files are included / excluded? It seems like a possible point of confusion that'd be tricky to debug.

Reply:
For the directory export they are preserved, for the archive export no. Given that all of the hosting providers we work with are using the directory export/git remote, and the timestamp of a file is inherently a local attribute I don't think this is an issue? Happy to switch the archive command out if you've got an example otherwise though :+1:

I've documented the behaviour with .gitattributes - that's actually kind of cool and potentially useful!